### DISCUSSION TITLE
How fix ModuleNotFoundError?

### DISCUSSION THREAD

**Discussion by finish475** [touhidurrr/iplist-youtube#2] [Q&A] [answered]
Traceback (most recent call last):
  File "C:\Users\admin\Downloads\list_generator.py", line 4, in <module>
    from dns import resolver
ModuleNotFoundError: No module named 'dns'

I'm not a programmer. How can I run it on Openwrt router. I am in Vietnam so many IP addresses have not been added to the list

**Comment by touhidurrr** [comment]
To use the Python Code, you need to install a Python Module called [dnspython](https://www.dnspython.org/) first. To install it run the following command:
```py
pip3 install -I dnspython==2.1.0
```
After installation your issue should be resolved.

Also, note that, Google use DNS based traffic routing (as far as I see), so fetching the Queries only one time might not be enough. You have to fetch them multiple times with an interval of at least 5 miniutes (Google’s TTL) to get all or most of the IP's they use in your location.

**Reply by finish475** [reply]
Thanks !

**Comment by touhidurrr** [comment, marked as answer]
Update on the discussion, use the following command to install all dependencies from `requirements.txt`:
```bash
pip3 install -r requirements.txt
```
